src/battle/overlay_12_0224E4FC.c

```diff
@@ -5541,3 +5542,939 @@ BOOL TryFling(BattleSystem *bsys, BATTLECONTEXT *ctx, int battlerId) {
     
     return TRUE;
 }
+
+void ov12_022565E0(BattleSystem *bsys, BATTLECONTEXT *ctx) {
+    if (GetBattlerHeldItemEffect(ctx, ctx->battlerIdAttacker) == HOLD_EFFECT_BOOST_REPEATED) {
+        if (!(ctx->battleMons[ctx->battlerIdAttacker].status2 & STATUS2_RAGE) &&
+            !(ctx->battleMons[ctx->battlerIdAttacker].status2 & STATUS2_UPROAR) &&
+            !(ctx->linkStatus & (1 << 9)) &&
+            !(ctx->battleMons[ctx->battlerIdAttacker].status2 & STATUS2_LOCKED_INTO_MOVE)) {
+            if (ctx->moveNoMetronome[ctx->battlerIdAttacker] == ctx->moveNoTemp) {
+                if (ctx->battleMons[ctx->battlerIdAttacker].unk88.metronomeTurns < 10) {
+                    ctx->battleMons[ctx->battlerIdAttacker].unk88.metronomeTurns++;
+                }
+            } else {
+                ctx->battleMons[ctx->battlerIdAttacker].unk88.metronomeTurns = 0;
+                ctx->moveNoMetronome[ctx->battlerIdAttacker] = ctx->moveNoTemp;
+            }
+        }
+    } else {
+        ctx->battleMons[ctx->battlerIdAttacker].unk88.metronomeTurns = 0;
+    }
+}
+
+void ov12_02256694(BattleSystem *bsys, BATTLECONTEXT *ctx) {
+    if (GetBattlerHeldItemEffect(ctx, ctx->battlerIdAttacker) == HOLD_EFFECT_BOOST_REPEATED) {
+        if ((ctx->moveStatusFlag & MOVE_STATUS_FAIL) &&
+            ctx->moveNoMetronome[ctx->battlerIdAttacker] == ctx->moveNoTemp &&
+            ctx->battleMons[ctx->battlerIdAttacker].unk88.metronomeTurns &&
+            !(ctx->selfTurnData[ctx->battlerIdAttacker].rolloutCount) &&
+            !(ctx->battleMons[ctx->battlerIdAttacker].status2 & STATUS2_RAGE) &&
+            !(ctx->battleMons[ctx->battlerIdAttacker].status2 & STATUS2_UPROAR) &&
+            !(ctx->linkStatus & (1 << 9)) &&
+            !(ctx->battleMons[ctx->battlerIdAttacker].status2 & STATUS2_LOCKED_INTO_MOVE)) {
+            ctx->battleMons[ctx->battlerIdAttacker].unk88.metronomeTurns--;
+        }
+    } else {
+        ctx->battleMons[ctx->battlerIdAttacker].unk88.metronomeTurns = 0;
+    }
+}
+
+//Related to send out Pokemon crys..?
+int ov12_02256748(BATTLECONTEXT *ctx, int battlerId, int battlerType, BOOL encounter) {
+    int ret;
+    int color;
+    BOOL half;
+    
+    if (encounter == TRUE && (battlerType == 2 || battlerType == 3)) {
+        half = TRUE;
+    } else {
+        half = FALSE;
+    }
+
+    ret = 0;
+    
+    if (half == TRUE) {
+        ret = 0;
+    }
+    
+    color = sub_020880B0(ctx->battleMons[battlerId].hp, ctx->battleMons[battlerId].maxHp, 48);
+    
+    if ((ctx->battleMons[battlerId].status & STATUS_ALL) || (color != 4 && color != 3)) {
```

Comment:
color constans? are you sure this is actually color?

Reply:
Im like 30% sure

Reply:
I think the colour specifically is what the HP bar is currently displaying, so not sure what each one would be on a scale of full health green to critical health red